translator/translate/otel/receiver/otlp/translator.go

```diff
@@ -77,26 +85,48 @@ func (t *translator) ID() component.ID {
 
 func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
 	cfg := t.factory.CreateDefaultConfig().(*otlpreceiver.Config)
-
-	// TODO: Should follow pattern done in awsemf and awsexray exporter translations (i.e should be integrated with standard otlp translation)
+	// init default configuration
+	configKeyExt := common.OtlpKey
+	cfg.GRPC.NetAddr.Endpoint = defaultGrpcEndpoint
+	cfg.HTTP.Endpoint = defaultHttpEndpoint
 	if t.name == common.AppSignals {
-		return common.GetYamlFileToYamlConfig(cfg, appSignalsConfig)
+		configKeyExt = common.AppSignals
+		cfg.GRPC.NetAddr.Endpoint = defaultAppSignalsGrpcEndpoint
+		cfg.HTTP.Endpoint = defaultAppSignalsHttpEndpoint
 	}
-
-	configKey, ok := configKeys[t.dataType]
+	configKeyBase, ok := configKeys[t.dataType]
 	if !ok {
 		return nil, fmt.Errorf("no config key defined for data type: %s", t.dataType)
 	}
+	configKey := common.ConfigKey(configKeyBase, configKeyExt)
 	if conf == nil || !conf.IsSet(configKey) {
 		return nil, &common.MissingKeyError{ID: t.ID(), JsonKey: configKey}
 	}
-	cfg.GRPC.NetAddr.Endpoint = defaultGrpcEndpoint
-	cfg.HTTP.Endpoint = defaultHttpEndpoint
-	if endpoint, ok := common.GetString(conf, common.ConfigKey(configKey, "grpc_endpoint")); ok {
-		cfg.GRPC.NetAddr.Endpoint = endpoint
+
+	var otlpKeyMap map[string]interface{}
+	if otlpSlice := common.GetArray[any](conf, configKey); t.index != -1 && len(otlpSlice) > t.index {
+		otlpKeyMap = otlpSlice[t.index].(map[string]interface{})
+	} else {
+		otlpKeyMap = conf.Get(configKey).(map[string]interface{})
+	}
+	if t.name == common.AppSignals {
```

Comment:
There is some conditional logic for AppSignals further up already. Can we keep it all in one place?